packages/xchain-ethereum/src/client.ts

```diff
@@ -343,12 +344,16 @@ export default class Client implements XChainClient, EthereumClient {
       if (account.tokens) {
         account.tokens.forEach((token) => {
           const decimals = parseInt(token.tokenInfo.decimals)
-          const tokenAsset = assetFromString(`${ETHChain}.${token.tokenInfo.symbol}-${token.tokenInfo.address}`)
-          if (tokenAsset && getTokenAddress(tokenAsset)) {
-            balances.push({
-              asset: tokenAsset,
-              amount: baseAmount(token.balance, decimals),
-            })
+          const symbol = token.tokenInfo.symbol
+          const tokenAddress = token.tokenInfo.address
+          if (validateSymbol(symbol) && this.validateAddress(tokenAddress)) {
+            const tokenAsset = assetFromString(`${ETHChain}.${symbol}-${tokenAddress}`)
+            if (tokenAsset && getTokenAddress(tokenAsset)) {
```

Comment:
Check of `getTokenAddress(tokenAsset)` is redundant, if we have validated `symbol` + `tokenAddress` before, because `assetFromString` should just work then.